ale/drivers/co_drivers.py

```diff
@@ -110,6 +110,186 @@
+        Therefore they use the same NAIF ID asin bods2c calls. Expects instrument_id to be
+        defined from a mixin class. This should return a string containing either 'ISSNA' or
+        'ISSWA'
+
+        Returns
+        -------
+        : str
+          instrument id
+        """
+        return id_lookup[super().instrument_id]
+
+    @property
+    def spacecraft_name(self):
+        """
+        Spacecraft name used in various Spice calls to acquire
+        ephemeris data.
+
+        Returns
+        -------
+        : str
+          Name of the spacecraft
+        """
+        return 'CASSINI'
+
+    @property
+    def sensor_name(self):
+        """
+        Returns the name of the instrument
+
+        Returns
+        -------
+        : str
+          Name of the sensor
+        """
+        return name_lookup[super().instrument_id]
+
+    @property
+    def ephemeris_start_time(self):
+        """
+        Returns the start and stop ephemeris times for the image.
+
+        Returns
+        -------
+        : float
+          start time
+        """
+        return spice.str2et(self.utc_start_time.strftime("%Y-%m-%d %H:%M:%S.%f"))[0]
+
+    @property
+    def center_ephemeris_time(self):
+        """
+        Returns the starting ephemeris time as the ssi framers center is the
+        start.
+
+        Returns
+        -------
+        : double
+          Center ephemeris time for an image
+        """
+        center_time = self.ephemeris_start_time + (self.exposure_duration / 2.0)
+        return center_time
+
+    @property
+    def odtk(self):
+        """
+        The radial distortion coeffs are not defined in the ik kernels, instead
+        they are defined in the ISS Data User Guide (Knowles). Therefore, we
+        manually specify the codes here.
+        Expects instrument_id to be defined. This should be a string containing either
+        CASSINI_ISS_WAC or CASSINI_ISIS_NAC
+
+        Returns
+        -------
+        : list<float>
+          radial distortion coefficients
+        """
+        if self.instrument_id == 'CASSINI_ISS_WAC':
+            # WAC
+            return [0, float('-6.2e-5'), 0]
+        elif self.instrument_id == 'CASSINI_ISS_NAC':
+            # NAC
+            return [0, float('-8e-6'), 0]
+
+    @property
+    def focal_length(self):
+        """
+        NAC uses multiple filter pairs, each filter combination has a different focal length.
+        NAIF's Cassini kernels do not contain focal lengths for NAC filters and
+        so we aquired updated NAC filter data from ISIS's IAK kernel.
+
+        """
+        # default focal defined by IAK kernel
+        try:
+            default_focal_len = super(CassiniIssPds3LabelNaifSpiceDriver, self).focal_length
+        except:
+            default_focal_len = float(spice.gdpool('INS{}_DEFAULT_FOCAL_LENGTH'.format(self.ikid), 0, 2)[0])
+
+        filters = tuple(self.label["IsisCube"]["BandBin"]['FilterName'].split("/"))
```

Comment:
since this can still git the spice kernels, should we start cacheing the values with an underscore prefixed variable? Or maybe use the naif keywords property and have that be cached in the background.

Reply:
Yes, I think anything that uses a spice call should probably be cached in an underscore